crates/engine/tree/src/tree/mod.rs

```diff
@@ -2502,6 +2458,10 @@ where
 
         // emit insert event
         let elapsed = start.elapsed();
+        // A block is considered a fork if its height is less than or equal to the current canonical
+        // head height
+        let is_fork =
+            executed.recovered_block().number() <= self.state.tree_state.canonical_block_number();
```

Comment:
@mediocregopher @shekhirin  I believe  in theory this might not detect all fork blocks

for example if we already have an alternative block for the same canonical_block_number height buffered and this block is + 1

so perhaps we do need walk up the chain in the state buffer until canonical height for this until we reach the current_canonical_head height?

which is similar to to the removed is_fork function but should just be this loop

https://github.com/paradigmxyz/reth/blob/6fafff5f145f99bb8cccf3e1b88ea009955ab648/crates/engine/tree/src/tree/mod.rs#L943-L955

and use the buffered blocks, because we've executed the block we know it connects back to the canonical chain